cmake/machinecfg/RZAnsel.cmake

```diff
@@ -34,34 +34,65 @@ set(RZANSEL_ARCH "ppc64le")
 # NOTE: When updating dependencies with new compilers or packages, you should
 # ideally only need to update these variables to change the default behavior.
 
-set(RZANSEL_VIEW_DATE_LATEST "2020-12-17")
+set(RZANSEL_VIEW_DATE_LATEST "2021-01-04")
 set(RZANSEL_GCC_PREFERRED "GCC8")
 set(RZANSEL_COMPILER_PREFERRED "GCC")
 set(RZANSEL_GCC8_VERSION "8.3.1")
-
+set(RZANSEL_VARIANT_PREFIX "_no-mpi")
 set(RZANSEL_MPI_DATE "2020.08.19")
 set(MPIEXEC_EXECUTABLE "/usr/tcetmp/bin/jsrun" CACHE STRING "Mpi executable to use on RZansel system")
 set(RZANSEL_CUDA_VERSION "10.1.243")
 
-set(RZANSEL_CUDA_DEFAULT ON)
+if (NOT MACHINE_VARIANT) 
```

Comment:
Why not just make this an option instead of having to do this string parsing. `RZANSEL_MPI` in addition to `RZANSEL_CUDA`?

Reply:
Phil already set a precedent I wanted to be consistent with what had already done for Summit. Moving forward at some point I would prefer to standardize the options on different systems so it's less confusing for users.

Reply:
I feel like CMake has a perfectly good mechanism for specifying boolean options, so it's weird to replace that with a hyphen-separated list system.

Reply:
Ok, I don't really care, I just want it to be consistent. @pgrete what do you think?

Reply:
@pgrete

Reply:
My original intention for this was flexibility, e.g., when also changing and I probably didn't want to introduce more variable.
I don't really feel very strong about this.
From my point of view these machine files are suggestions and/or help to jumpstart a new user on a machine.
Eventually, I see everyone using their own machine file for downstream codes anyway (so different machine files can use different/custom logic and there's no right or wrong in writing them)

Reply:
@pgrete I would agree with that statement from the point of view of user specefic machine config files, but I feel like we should have a slightly higher standard for machine config files we actually maintain. Regardless this is not a priority. @AndrewGaspar if we are not going to standardize this, I don't think it is worth changing.

Reply:
Trying on Summit I remember now why I made that choice for a single parameter: convenience/laziness (I only wanted to change one variable on the command line rather than one variable per option)

And now that I think even more about this, I'm wondering why, e.g.,  the `RZANSEL_CUDA` or `RZANSEL_CXX_COMPILER` variables are required at all (instead of setting `Kokkos_ENABLE_CUDA` or `CMAKE_CXX_COMPILER`, respectively).
From my point of view the main purpose of the machine files is to set reasonable defaults for a specific machine. For this reason, those defaults (the values) can be very machine specific (e.g., full path to compiler etc) but don't necessarily need to/should be new variables (if the required functionality is already provided by Parthenon default variables).

Bottom line, I'm now in favor of drastically simplifying the machine file (if you like I can give it a shot).

Reply:
I think we should continue this conversation on this issue. https://github.com/lanl/parthenon/issues/372

Reply:
`RZANSEL_CXX_COMPILER` isn't a user option. It's just a variable to store the host compiler.

The reason I made `RZANSEL_CUDA` separate from `Kokkos_ENABLE_CUDA` is because at some point my intention was to actually just build out Kokkos in the spack environment as well, in which case `Kokkos_ENABLE_CUDA` would do nothing to Kokkos and merely control which environment you use in RZAnsel.

What I didn't want to do was make the machine configuration depend on a bunch of variables which aren't defined (along with their defaults) until later in the CMake configuration. So I figured I'd scope the variable names to the machine.